Formula/qpid-proton.rb

```diff
@@ -0,0 +1,64 @@
+class QpidProton < Formula
+  desc "is a high-performance, lightweight AMQP 1.0 messaging library."
+  homepage "https://qpid.apache.org/proton/"
+  url "https://www-eu.apache.org/dist/qpid/proton/0.17.0/qpid-proton-0.17.0.tar.gz"
+  sha256 "6ffd26d3d0e495bfdb5d9fefc5349954e6105ea18cc4bb191161d27742c5a01a"
+
+  depends_on "cmake" => :build
+  depends_on "libuv"
+  depends_on "openssl"
+  depends_on "perl" if Formula["perl"].installed?
```

Comment:
Sorry, we don't support that use-case. `depends_on :perl` does what you suggest if they are building from source and uses the Homebrew `perl` when using a binary package.